unit/util/range.cpp

```diff
@@ -65,3 +65,95 @@ SCENARIO("range tests", "[core][util][range]")
     }
   }
 }
+
+class move_onlyt
+{
+public:
+  move_onlyt(move_onlyt &&) = default;
+  move_onlyt &operator=(move_onlyt &&) = default;
+  move_onlyt(const move_onlyt &) = delete;
+  move_onlyt &operator=(const move_onlyt &) = delete;
+
+  explicit move_onlyt(int value) : value{value} {};
```

Comment:
:pick: I'd have probably done an actually move-only type (e.g. with a `std::unique_ptr`) so I'd be sure that I'd really fixed it.